advisor-backend/routes/schedule.js

```diff
@@ -658,4 +783,245 @@ router.post("/create", async (req, res) => {
   }
 });
 
+// this route handles course swaps into years that are not full of classes already
+router.post("/nonfull", (req, res) => {
```

Comment:
I think it would make more sense for all of the swaps to have a single endpoint, and you could post this metadata as a part of the input to that endpoint, or determine it from the schedule instead.